zwave_js_server/client.py

```diff
@@ -2,95 +2,51 @@
 STATE_CONNECTED = "connected"
 STATE_DISCONNECTED = "disconnected"
 
-
 SIZE_PARSE_JSON_EXECUTOR = 8192
 
 
-async def gather_callbacks(
-    logger: logging.Logger, name: str, callbacks: List[Callable[[], Awaitable[None]]]
-) -> None:
-    """Gather callbacks."""
-    results = await asyncio.gather(*[cb() for cb in callbacks], return_exceptions=True)
-    for result, callback in zip(results, callbacks):
-        if not isinstance(result, Exception):
-            continue
-        logger.error("Unexpected error in %s %s", name, callback, exc_info=result)
-
-
-class NotConnected(Exception):
-    """Exception raised when trying to handle unknown handler."""
-
-
-class InvalidState(Exception):
-    """Exception raised when data gets in invalid state."""
-
-
-class InvalidServerVersion(Exception):
-    """Exception raised when connected to server with incompatible version."""
-
-
-class FailedCommand(Exception):
-    """When a command has failed."""
-
-    def __init__(self, message_id: str, error_code: str):
-        """Initialize a failed command error."""
-        super().__init__(f"Command failed: {error_code}")
-        self.message_id = message_id
-        self.error_code = error_code
-
-
 class Client:
     """Class to manage the IoT connection."""
 
-    def __init__(
-        self,
-        ws_server_url: str,
-        aiohttp_session: ClientSession,
-        *,
-        start_listening_on_connect: bool = True,
-    ):
+    def __init__(self, ws_server_url: str, aiohttp_session: ClientSession):
         """Initialize the Client class."""
         self.ws_server_url = ws_server_url
         self.aiohttp_session = aiohttp_session
         self.driver: Optional[Driver] = None
         # The WebSocket client
         self.client: Optional[ClientWebSocketResponse] = None
-        # Scheduled sleep task till next connection retry
-        self.retry_task: Optional[asyncio.Task] = None
-        # Boolean to indicate if we wanted the connection to close
-        self.close_requested = False
-        # The current number of attempts to connect, impacts wait time
-        self.tries = 0
         # Current state of the connection
         self.state = STATE_DISCONNECTED
         # Version of the connected server
         self.version: Optional[VersionInfo] = None
-        self._on_connect: List[Callable[[], Awaitable[None]]] = []
-        self._on_disconnect: List[Callable[[], Awaitable[None]]] = []
-        self._on_initialized: List[Callable[[], Awaitable[None]]] = []
         self._logger = logging.getLogger(__package__)
-        self._disconnect_event: Optional[asyncio.Event] = None
-        self._result_futures: Dict[str, asyncio.Future] = {}
         self._loop = asyncio.get_running_loop()
-
-        if start_listening_on_connect:
-            self.register_on_connect(self._start_listening)
+        self._result_futures: Dict[str, asyncio.Future] = {}
 
     def __repr__(self) -> str:
         """Return the representation."""
```

Comment:
Should `async_handle_message` be `_async_handle_message` ?

Reply:
Yeah, probably. I also think we should remove the `async_` prefix from the client methods.